HW2Question2.cpp

```diff
@@ -0,0 +1,91 @@
+//
+//  HW2Question2.cpp
+//  Codu
+//
+//  Created by Farida Eid on 5/27/17.
+//  Copyright © 2017 Farida Eid. All rights reserved.
+//
+
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+struct treeNode
+{
+    int value;
+    treeNode *left;
+    treeNode *right;
+    
+    treeNode(int v):value(v), left(NULL), right(NULL){}
+    
+};
+
+// find path of each node and checks if the node is in the tree or not
+
+bool findPath(treeNode * head, int key, vector<int>& answer)
+{
+    if(head == NULL) return false;
+    answer.push_back(head->value);      // need it to stop the loop in LCA functoin
+    if(head->value == key) return true;
+    
+    if(findPath(head->left, key, answer) || findPath(head->right, key, answer))
+        return true;
+    
+    answer.pop_back();
+    return false;
+}
+
+int LCA(treeNode *head, int key1, int key2)
```

Comment:
Please use a full name.
Also, add a comment to the function.
It's a bit borderline, but the problem statement says 'Avoid storing additional nodes in a data structure.', storing keys to the nodes in a vector is kind of the same. I think that is to avoid this solution :), if you can, please think of another solution.